### PR TITLE
Add GKE integration pipeline

### PR DESCRIPTION
**Pull request by jakkab**
**Description**

Changes proposed in this pull request:

- add GKE integration pipeline

**Related issue(s)**
#20

### CODE REVIEW

File: gke-integration-job/pr-gke-integration.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+ROOT_PATH="/home/prow/go/src/github.com/kyma-project/kyma"
+
+echo "
+################################################################################
+# Provisioning GKE cluster
+################################################################################
+"
+
+CLUSTER_NAME="${REPO_OWNER}-${REPO_NAME}-${PULL_NUMBER}" bash ${ROOT_PATH}/prow/scripts/provision-gke-cluster.sh
```
Comment (Tomasz-Smelcerz-SAP) [inline gke-integration-job/pr-gke-integration.sh]: Add deprovisioning

File: gke-integration-job/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM eu.gcr.io/kyma-project/prow/bootstrap:0.0.1
```
Comment (aszecowka) [inline gke-integration-job/Dockerfile]: IMO we should use prow/images directory

Comment (aszecowka) [reply gke-integration-job/Dockerfile]: but I doubt if we really need to create image: see changes proposed by Suleyman: https://github.com/kyma-project/test-infra/pull/98/files#diff-4, he gets rid of creating image and use bootrstap with mounted test-infra repo.

File: prow/config.yaml [hunk 3]
```diff
@@ -12,6 +12,23 @@ presets:
       - name: sa-vm-kyma-integration
         mountPath: /etc/credentials/sa-vm-kyma-integration
         readOnly: true
+  - labels:
+      preset-sa-gke-kyma-integration: "true"
+    env:
+      - name: GCLOUD_SERVICE_KEY_PATH
+        value: /etc/credentials/sa-gke-kyma-integration/service-account.json
+      - name: GCLOUD_PROJECT_NAME
```
Comment (aszecowka) [inline prow/config.yaml]: there will be preset introduced by Suleyman's PR: https://github.com/kyma-project/test-infra/pull/98/files#diff-4

File: prow/jobs/kyma/kyma-integration.yaml [hunk 4]
```diff
@@ -19,19 +19,19 @@ presubmits: # runs on PRs
...
       spec:
         containers:
-          - image: eu.gcr.io/kyma-project/prow/bootstrap:0.0.1
+          - image: eu.gcr.io/kyma-project/snapshot/test/gke-integration:0.0.3 # created by running `docker build -t <image> .` in the gke-integration-job directory.
             securityContext:
               privileged: true
-            command: ["ls"]
-            args: ["-la"]
+            command: ["bash"]
+            args: ["/pr-gke-integration.sh"]
```
Comment (aszecowka) [inline prow/jobs/kyma/kyma-integration.yaml]: remember to configure postsubmit job

**Comment by Tomasz-Smelcerz-SAP** [comment]
I am closing this pull request because required changes are too big. This is continued in #100

**Pull request closed without merging**